app/internal/export.py

```diff
@@ -0,0 +1,123 @@
+from datetime import datetime
+from typing import List
+
+import pytz
+from icalendar import Calendar, vCalAddress, vText
+from icalendar import Event as IcalEvent
+from sqlalchemy.orm import Session
+
+from app.config import DOMAIN, ICAL_VERSION, PRODUCT_ID
+from app.database.models import Event
+from app.routers.event import get_attendees_email
+
+
+def generate_id(event: Event) -> bytes:
+    """Creates an unique id."""
+
+    return (
+            str(event.id)
+            + event.start.strftime('%Y%m%d')
+            + event.end.strftime('%Y%m%d')
+            + f'@{DOMAIN}'
+    ).encode()
+
+
+def create_ical_calendar():
+    """Creates an ical calendar,
+    and adds the required information"""
+
+    cal = Calendar()
+    cal.add('version', ICAL_VERSION)
+    cal.add('prodid', PRODUCT_ID)
+
+    return cal
+
+
+def add_optional(user_event, data):
+    """Adds an optional field if it exists."""
+
+    if user_event.location:
+        data.append(('location', user_event.location))
+
+    if user_event.content:
+        data.append(('description', user_event.content))
+
+    return data
+
+
+def create_ical_event(user_event):
+    """Creates an ical event,
+    and adds the event information"""
+
+    ievent = IcalEvent()
+    data = [
+        ('organizer', add_attendee(user_event.owner.email, organizer=True)),
+        ('uid', generate_id(user_event)),
+        ('dtstart', user_event.start),
+        ('dtstamp', datetime.now(tz=pytz.utc)),
+        ('dtend', user_event.end),
+        ('summary', user_event.title),
+    ]
+
+    data = add_optional(user_event, data)
+
+    for param in data:
+        ievent.add(*param)
+
+    return ievent
+
+
+def add_attendee(email, organizer=False):
```

Comment:
Please add type annotations and check the validity of the `email` parameter

Reply:
isn't it better to validate the `email `on the `FE`?